### PR TITLE
create private router

### LINKED ISSUE #7: Create a private router

### PR DESCRIPTION
**Pull request by IsaiasSantanaDosSantos**
Closed #7

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
     "@testing-library/jest-dom": "^5.16.5",
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
-    "history": "^5.3.0",
+    "history": "^4.10.1",
```
Comment (IsaiasSantanaDosSantos) [inline package.json:9]: Manter v4

**Review by IsaiasSantanaDosSantos — COMMENTED** [review]
All Right now

File: src/routes/index.js [hunk 2]
```diff
@@ -1,14 +1,15 @@
 import React from "react";
-import { Switch, Route } from "react-router-dom";
```
Comment (IsaiasSantanaDosSantos) [inline src/routes/index.js:2]: oi

File: src/routes/index.js [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
+import MyRoute from "./MyRoute";
 import Login from "../pages/Login";
 import Page404 from "../pages/Page404";
 
 export default function Routes() {
   return (
     <Switch>
-      <Route exact path="/" component={Login} />
-      <Route path="*" component={Page404} />
+      <MyRoute exact path="/" component={Login} />
```
Comment (IsaiasSantanaDosSantos) [inline src/routes/index.js:11]: ok

**Review by IsaiasSantanaDosSantos — COMMENTED** [review]
ok

File: .prettierrc.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+// printWidth = 100;
+// singleQuote = false;
+// trailingComma = "es5";
+// jsxBracketSameLine = true;
+module.exports = {
+  semi: true,
+  tabWidth: 2,
+  printWidth: 100,
+  singleQuote: false,
+  trailingComma: "es5",
```
Comment (IsaiasSantanaDosSantos) [inline .prettierrc.js:12]: ok

File: src/routes/MyRoute.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  return <Route {...rest} component={Component} />;
+}
+//protp
+MyRoute.defaultProps = {
+  isClosed: false,
+};
+
+MyRoute.propTypes = {
+  component: PropTypes.oneOfType([PropTypes.element, PropTypes.func]).isRequired,
+  isClosed: PropTypes.bool,
```
Comment (IsaiasSantanaDosSantos) [inline src/routes/MyRoute.js:21]: ok

**Review by IsaiasSantanaDosSantos — COMMENTED** [review]
ok

File: package.json [hunk 6]
```diff
@@ -6,7 +6,8 @@
...
+    "prop-types": "^15.8.1",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-icons": "^4.8.0",
```
Comment (IsaiasSantanaDosSantos) [inline package.json:13]: ok

**Review by IsaiasSantanaDosSantos — COMMENTED** [review]
ok

**Review by IsaiasDeveloper — APPROVED** [review]
Now, all right. I hope!

**Review by IsaiasSantanaDosSantos — COMMENTED** [review]
ok

**Review by IsaiasSantanaDosSantos — COMMENTED** [review]
Okay

**Pull request merged**